internal/venice-common/src/main/java/com/linkedin/venice/writer/update/UpdateBuilderImpl.java

```diff
@@ -130,6 +135,10 @@ public GenericRecord build() {
   private Exception validateUpdateRecordIsSerializable(GenericRecord updateRecord) {
     try {
       serializer.serialize(updateRecord);
+    } catch (UnresolvedUnionException serializationException) {
+      Object unresolvedDatum = serializationException.getUnresolvedDatum();
+      LOGGER.error("Unresolved datum encountered: {}", unresolvedDatum, serializationException);
+      return serializationException;
```

Comment:
Ok so... I re-read the context of the issue that led us to create this ticket, and I remember it better now... the problem is that although the `UnresolvedUnionException` already prints the `unresolvedDatum` in its exception message, that is not always sufficient to debug the issue. We would also want to print the type/schema of that datum. So I would suggest something along those lines instead:

```
      Object unresolvedDatum = serializationException.getUnresolvedDatum();
      String unresolvedDatumType = unresolvedDatum instanceof GenericContainer
          ? ((GenericContainer) unresolvedDatum).getSchema().toString()
          : unresolvedDatum.getClass().getSimpleName();
      return new VeniceException("The following type does not conform to any branch of the union: " + unresolvedDatumType, serializationException);
```

Since the code calling `validateUpdateRecordIsSerializable` will throw the exception, and that should eventually get printed higher up in the stack, I don't think it's necessary to print it ourselves down here. Double-printing exceptions is sometimes confusing, so I think it's better to have all the info we need within the one exception.

Reply:
Ah I see. I added the changes let me know if it looks good.